app/models/school.rb

```diff
@@ -42,6 +43,27 @@ def partially_registered?
       &.any?
   end
 
+  def pupil_premium_uplift?(start_year)
+    pupil_premium_eligibilities.find_by(start_year: start_year)&.uplift? || false
+  end
+
+  scope :with_pupil_premium_uplift, lambda { |start_year|
+    query = <<~QUERY
+      pupil_premium_eligibilities.start_year = ?
+      AND (
+        pupil_premium_eligibilities.percent_primary_pupils_eligible >= ?
+        OR pupil_premium_eligibilities.percent_secondary_pupils_eligible >= ?
+      )
+    QUERY
+    joins(:pupil_premium_eligibilities)
+      .where(
+        query,
+        start_year,
+        PupilPremiumEligibility::THRESHOLD_PERCENTAGE,
+        PupilPremiumEligibility::THRESHOLD_PERCENTAGE,
+      )
```

Comment:
I wonder, is this part of the scope placed appropriately? 

Or should we make a scope for it on `PupilPremiumEligibility`? 

`only_uplifting_premiums` or something along those lines?

Reply:
Also, I know you are gonna love it, buut...
Do we want a test for `with_pupil_premium_uplift`?

Reply:
No.

Reply:
😛

Reply:
Done the first part, working on tests